CMakeLists.txt

```diff
@@ -1,16 +1,36 @@
 cmake_minimum_required(VERSION 3.12)
 project(Boost-CMake)
 
-option(BOOST_DOWNLOAD_TO_BINARY_DIR "Prefer downloading Boost to the binary directory instead of source directory" OFF)
 option(BOOST_DISABLE_TESTS "Do not build test targets, even if building standalone" OFF)
+option(BOOST_USE_GIT "Get Boost from git source" OFF)
 
 set(BOOST_URL "https://dl.bintray.com/boostorg/release/1.69.0/source/boost_1_69_0.tar.bz2" CACHE STRING "Boost download URL")
 set(BOOST_URL_SHA256 "8f32d4617390d1c2d16f26a27ab60d97807b35440d45891fa340fc2648b04406" CACHE STRING "Boost download URL SHA256 checksum")
-set(BOOST_ARCHIVE_DIRECTORY "" CACHE DIRECTORY "Use the specified local directory to search for boost archives instead of downloading them" )
 
-list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
+include(FetchContent)
+
+if(BOOST_USE_GIT)
+  FetchContent_Declare(
+    Boost
+    GIT_REPOSITORY "https://github.com/boostorg/boost.git"
+    GIT_TAG "boost-1.69.0"
```

Comment:
Would be nice to have the git version specified in 1 place, and `string()` code set up to format it as needed in the URL and Git repo cases. This way when you upgrade to a new version of Boost, you aren't changing 3-4+ places

Reply:
Done. Note that you can always use FetchContent_Declare in the global CMake project before using add_subdirectory() on boost-cmake to chose the branch or tag you prefer.

Still, I moved those strings to option() as it can be more convenient.

Reply:
I apologize for the confusion, I didn't mean a configuration setting. I meant just for maintainability, a "local" variable to store it to make it easier for the maintainers to update the version later. In the case where the user wants to override the version, they must also specify a tag or SHA1, so 1 variable doesn't quite cut it. In that case, yeah I agree, they need to use Declare.